### PR TITLE
DRILL-1942-hygiene:

### PR DESCRIPTION
**Pull request by cwestin**
DRILL-1942-hygiene (another preparatory patch):
    - add AutoCloseable to some classes
    - minor fixes
    - formatting

Unit tests pass
Regression suite passed except for some window function queries (this branch needs to get rebased to pick up that code), and the known fragment distribution timeout.

### CODE REVIEW

File: exec/java-exec/src/main/java/org/apache/drill/exec/ops/BufferManager.java [hunk 1]
```diff
@@ -51,11 +48,12 @@ public BufferManager(BufferAllocator allocator, FragmentContext fragmentContext)
...
 
+  @Override
   public void close() throws Exception {
-    Object[] mbuffers = ((LongObjectOpenHashMap<Object>)(Object)managedBuffers).values;
-    for (int i =0; i < mbuffers.length; i++) {
+    final Object[] mbuffers = ((LongObjectOpenHashMap<Object>) (Object) managedBuffers).values;
+    for (int i = 0; i < mbuffers.length; i++) {
       if (managedBuffers.allocated[i]) {
-        ((DrillBuf)mbuffers[i]).release();
+        ((DrillBuf) mbuffers[i]).release(1);
```
Comment (adeneche) [inline exec/java-exec/src/main/java/org/apache/drill/exec/ops/BufferManager.java]: why not just call release(), it calls release(1) internally ?

File: exec/java-exec/src/main/java/org/apache/drill/exec/ops/BufferManager.java [hunk 2]
```diff
@@ -65,7 +63,7 @@ public DrillBuf replace(DrillBuf old, int newSize) {
     if (managedBuffers.remove(old.memoryAddress()) == null) {
       throw new IllegalStateException("Tried to remove unmanaged buffer.");
     }
-    old.release();
+    old.release(1);
```
Comment (adeneche) [inline exec/java-exec/src/main/java/org/apache/drill/exec/ops/BufferManager.java]: same as above

File: exec/java-exec/src/main/java/org/apache/drill/exec/ops/BufferManager.java (hunk 1 above)
Comment (cwestin) [inline exec/java-exec/src/main/java/org/apache/drill/exec/ops/BufferManager.java]: It saves a function call; seems desirable in tight loops and low-level value vector situations.

File: exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/join/HashJoinBatch.java [hunk 3]
```diff
@@ -481,30 +466,28 @@ public HashJoinProbe setupHashJoinProbe() throws ClassTransformationException, I
...
 
         fieldId++;
         outputFieldId++;
       }
-      recordCount = left.getRecordCount();
     }
 
-    HashJoinProbe hj = context.getImplementationClass(cg);
-
+    final HashJoinProbe hj = context.getImplementationClass(cg);
```
Comment (adeneche) [inline exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/join/HashJoinBatch.java]: you can also change this to:
```
return context.getImplementationClass(cg);
```

**Comment by adeneche** [comment]
+1 LGTM

**Pull request closed without merging**